soc/nxp/mcx/mcxw/mcxw71_platform_init.S

```diff
@@ -16,13 +16,11 @@
 #include <zephyr/linker/sections.h>
 
 _ASM_FILE_PROLOGUE
-#ifdef CONFIG_PLATFORM_SPECIFIC_INIT
 
+GTEXT(soc_reset_hook)
+SECTION_SUBSEC_FUNC(TEXT,_reset_section,soc_reset_hook)
 
-GTEXT(z_arm_platform_init)
-SECTION_SUBSEC_FUNC(TEXT,_reset_section,z_arm_platform_init)
-
-.z_arm_platform_init:
+.soc_reset_hook:
```

Comment:
I would suggest to have a dedicated function for the ECC ram init, and have `soc_reset_hook` calling it.
The reason behind is we may need to call the ECC ram init as a standalone procedure, outside the early init.
For instance, at wake up from power down mode, if an ECC ram bank has been shutdown, we need to go through this procedure again, so we could call it directly.

This is why in #79060, we implemented `soc_reset_hook` in C, and the ECC init function separately.
Having `soc_reset_hook` in C would likely be "easier" to maintain and extend if needed.

Reply:
Where is placed the stack ? STCM3 and STCM4 are non-ecc (range 0x30010000 - 0x30019FFF)

Reply:
and also, to answer your question, the stack can be anywhere, it just goes at the end of the image. Whether or not it goes in non-ecc range is just depending on the size of the image.